src/django_idom/hooks.py

```diff
@@ -64,19 +83,48 @@ def use_websocket() -> IdomWebsocket:
     return websocket
 
 
+@overload
 def use_query(
     query: Callable[_Params, _Result | None],
+    options: QueryOptions,
```

Comment:
From a Python perspective, this would feel wrong. PEP8 positional args should be "absolute" in positioning. So whatever is the first argument should always be the first arg.

```python
# Query is the first arg in a normal situation
use_query(query, arg1, arg2)

# Now query is the second arg, which means query's position is not absolute
use_query(options, query, arg1, arg2)
```

Reply:
One could make a similar argument about the fact that `arg1` must be shifted in order to add options after `query`. It seems to me that, given we're unable to use a kwarg to pass the options, this rule must be broken one way or another. With this in mind, I'd argue that have options go first is preferable because, with and without options, `query` and its options remain visually grouped.

Reply:
Also in the argument of arg1, that's a single violation. But moving both arg1 and query becomes two violations.

Reply:
This seems like a case where there should probably be one way to do things in order to aid in readability.

> But moving both arg1 and query becomes two violations.

I mean, if putting it first causes N violations, then putting it next causes N-1 (since you must shift `arg2` etc.) which I don't think is a big enough difference to counterbalance the value of grouping `query, *args, **kwargs`.

Reply:
I tried playing around with it, IMO the API feels very wrong to change to shift the query position around.

Since it's type hinted properly, I don't think there's any ambiguity about the second arg situation.

Reply:
I've moved the `QueryOptions` to the first arg